# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditor.h
# Commit: dfd4ec691465
# Full Hash: dfd4ec69146560b6de52edbc4acdb49762daf38c
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2023-01-17 16:13:02
# Regressor Bug: 1807829
# File Overlap Count: 2
# Description:
#   Bug 1807829 - Make `HTMLEditor::CreateStyleForInsertText` use same logic as `HTMLEditor::SetInlinePropertiesAsSubAction` to apply new style r=m_kato
#   
#   The problem in Yahoo mail is, when you apply new font size to:
#   ```
#   <div><font size="7">{}<br></font></div>
# ==============================================================================

diff -r 0f2913c81558 -r dfd4ec691465 editor/libeditor/HTMLEditor.h
--- a/editor/libeditor/HTMLEditor.h	Mon Jan 16 23:35:34 2023 +0000
+++ b/editor/libeditor/HTMLEditor.h	Mon Jan 16 23:42:27 2023 +0000
@@ -1079,10 +1079,12 @@
    * PendingStyles to proper element node.
    *
    * @param aPointToInsertText  The point to insert text.
+   * @param aEditingHost        The editing host.
    * @return                    A suggest point to put caret or unset point.
    */
   [[nodiscard]] MOZ_CAN_RUN_SCRIPT Result<EditorDOMPoint, nsresult>
-  CreateStyleForInsertText(const EditorDOMPoint& aPointToInsertText);
+  CreateStyleForInsertText(const EditorDOMPoint& aPointToInsertText,
+                           const Element& aEditingHost);
 
   /**
    * GetMostDistantAncestorMailCiteElement() returns most-ancestor mail cite
@@ -3267,6 +3269,16 @@
       const AutoTArray<EditorInlineStyleAndValue, N>& aStylesToSet);
 
   /**
+   * SetInlinePropertiesAroundRanges() applying the styles to the ranges even if
+   * the ranges are collapsed.
+   */
+  template <size_t N>
+  [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult SetInlinePropertiesAroundRanges(
+      AutoRangeArray& aRanges,
+      const AutoTArray<EditorInlineStyleAndValue, N>& aStylesToSet,
+      const Element& aEditingHost);
+
+  /**
    * RemoveInlinePropertiesAsSubAction() removes specified styles from
    * mPendingStylesToApplyToNewContent if `Selection` is collapsed.  Otherwise,
    * removing the style.